src/extension/php_objects.c

```diff
@@ -119,10 +149,86 @@ static zend_string *nullable _fetch_from_env(const char *name, size_t name_len)
     }
     *w = '\0';
 
+    return env_name;
+}
+
+#ifndef ZTS
+/**
+ * This function is based on fpm_php_zend_ini_alter_master
+ * from php-src
+ */
+static int _custom_php_zend_ini_alter_master(zend_ini_entry *nullable ini_entry,
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    zend_string *nullable new_value, int mode, int stage,
+    bool has_env) /* {{{ */
+{
+    zend_string *duplicate;
+
+    if (!ini_entry) {
+        return FAILURE;
+    }
+
+    duplicate = zend_string_dup(new_value, 1);
+
+    struct entry_ex *eex = ini_entry->mh_arg3;
+    eex->has_env = has_env;
+
+    if (!ini_entry->on_modify ||
+        ini_entry->on_modify(ini_entry, duplicate, ini_entry->mh_arg1,
+            ini_entry->mh_arg2, ini_entry->mh_arg3, stage) == SUCCESS) {
+        ini_entry->value = duplicate;
+        /* when mode == ZEND_INI_USER keep unchanged to allow ZEND_INI_PERDIR
+         * (.user.ini) */
+        if (mode == ZEND_INI_SYSTEM) {
+            ini_entry->modifiable = mode;
+        }
+    } else {
+        zend_string_free(duplicate);
+    }
+
+    return SUCCESS;
+}
+
+dd_result dd_phpobj_load_env_values()
+{
+    char *env_name = NULL;
+    zend_ini_entry *p;
+    zend_string *env_def;
```

Comment:
`env_name` and `env_def` can be defined inside the if below.

Reply:
Done